Assets/BossRoom/Scripts/Shared/NetworkNameState.cs

```diff
@@ -7,9 +7,25 @@ namespace BossRoom
     /// <summary>
     /// NetworkBehaviour containing only one NetworkVariableString which represents this object's name.
     /// </summary>
-    public class NetworkNameState : NetworkBehaviour
+    public class NetworkNameState : NetworkBehaviour, INetworkSubscribable<string>
```

Comment:
As discussed, netvars are meant to be used directly in gameplay. We shouldn't wrap them just to hide netvars from gameplay code. Because this is such a simple class here that's meant as a container for these values, it might not make sense to have encapsulation for this.
If you have other cases for it being dangerous to expose to users, please do tell, it might be a need for a refactor netvar side.